benchmarks/dynamo/dist_util.py

```diff
@@ -0,0 +1,147 @@
+import argparse
+import functools
+import importlib
+import os
+
+import torch
+import torch.distributed as dist
+import torch.nn as nn
+from torch._dynamo.testing import reduce_to_scalar_loss
+from torch.distributed.algorithms._checkpoint.checkpoint_wrapper import (
+    apply_activation_checkpointing,
+    checkpoint_wrapper,
+    CheckpointImpl,
+)
+from torch.distributed.fsdp import FullyShardedDataParallel as FSDP
+from torch.distributed.fsdp.wrap import transformer_auto_wrap_policy
+
+try:
+    from .torchbench import setup_torchbench_cwd
+except ImportError:
+    from torchbench import setup_torchbench_cwd
+
+
+def setup(rank, world_size):
+    os.environ["MASTER_ADDR"] = "localhost"
+    os.environ["MASTER_PORT"] = "12355"
+    dist.init_process_group("nccl", rank=rank, world_size=world_size)
+
+
+def cleanup():
+    dist.destroy_process_group()
+
+
+class CustomLinear(torch.nn.Module):
+    def __init__(self, a, b):
+        super(CustomLinear, self).__init__()
+        self.weight = nn.Parameter(torch.randn(a, b))
+
+    def forward(self, x):
+        return torch.mm(x, self.weight)
+
+
+class MyModule(torch.nn.Module):
+    def __init__(self, a, b):
+        super(MyModule, self).__init__()
+        self.net = nn.Sequential(
+            nn.Linear(a, b),
+            nn.ReLU(),
+        )
+
+    def forward(self, x):
+        return self.net(x)
+
+
+class ToyModel(nn.Module):
+    def __init__(self):
+        super(ToyModel, self).__init__()
+        self.net = nn.Sequential(
+            *[nn.Linear(10, 10000), nn.ReLU()]
+            + [nn.Linear(10000, 10000), nn.ReLU()]
+            + [MyModule(10000, 10000)]
+            + [MyModule(10000, 1000)]
+            + [MyModule(1000, 1000)]
+            + [MyModule(1000, 1000)]
+            + [MyModule(1000, 1000)]
+            + [MyModule(1000, 1000)]
+            + [MyModule(1000, 1000)]
+            + [MyModule(1000, 1000)]
+            + [MyModule(1000, 1000)]
+            + [nn.Linear(1000, 5)]
+        )
+
+    def forward(self, x):
+        return self.net(x)
+
+
+def model_iter_fn(model, example_inputs, collect_outputs=False):
+    outputs = model(*example_inputs)
+    loss = reduce_to_scalar_loss(outputs)
+    loss.backward()
+    if collect_outputs:
+        return outputs
+
+
+def get_model(args):
+    if args.torchbench_model:
+        old_cwd = setup_torchbench_cwd()
+        module = importlib.import_module(
+            f"torchbenchmark.models.{args.torchbench_model}"
```

Comment:
copying from torchdynamo's benchmark script, this script is runnable from the root of the pytorch repo as 
`python benchmarks/dynamo/distributed.py` and it takes care of 'cd'ing to torchbench inside the process if necessary.  Admittedly this is janky.  I've become used to it since it's the way benchmarks/dynamo/torchbench.py works too, so at least it is consistent.  IMO the only 'real' fix would be to modify torchbench so that it does not require you to be in a particular directory for it to work.  

as a small stopgap, i could add an argparser arg for `--torchbench_dir` which lets you tell it the rel/abs path to your torchbench.  however, once you learn how to use it the way it is you might find its ok.